module/zfs/qat_compress.c

```diff
@@ -464,19 +488,49 @@ qat_compress(qat_compress_dir_t dir, char *src, int src_len,
 	}
 
 	if (out_pages) {
-		for (page_num = 0;
-		    page_num < buf_list_dst->numBuffers;
-		    page_num++) {
+		for (page_num = 0; page_num < dst_pages; page_num++) {
 			kunmap(out_pages[page_num]);
 		}
 		QAT_PHYS_CONTIG_FREE(out_pages);
 	}
 
+	if (add_pages) {
+		for (page_num = 0;
+		    page_num < buf_list_dst->numBuffers - dst_pages;
+		    page_num++) {
+			kunmap(add_pages[page_num]);
+		}
+		QAT_PHYS_CONTIG_FREE(add_pages);
+	}
+
 	QAT_PHYS_CONTIG_FREE(buffer_meta_src);
 	QAT_PHYS_CONTIG_FREE(buffer_meta_dst);
 	QAT_PHYS_CONTIG_FREE(buf_list_src);
 	QAT_PHYS_CONTIG_FREE(buf_list_dst);
 
+	return (status);
+}
+
+/* Entry point for QAT accelerated compression / decompression. */
```

Comment:
nit: function header block comments should all be multi-line even they fit on one line.

```
/*
 *  Entry point for QAT accelerated compression / decompression
 */
```